### DISCUSSION TITLE
What exactly does "Can't use SSL_get_servername" mean?

### DISCUSSION THREAD

**Discussion by ckcr4lyf** [openssl/openssl#21523] [Q&A] [answered]
I am trying to manually connect to a HTTPS server, WITHOUT SNI. 

It works, but I would like to understand this "warning message" that comes up:

```
$ openssl s_client -connect 66.254.114.41:443
CONNECTED(00000003)
Can't use SSL_get_servername
depth=2 C = US, O = DigiCert Inc, OU = www.digicert.com, CN = DigiCert Global Root CA
verify return:1
depth=1 C = US, O = DigiCert Inc, CN = DigiCert TLS Hybrid ECC SHA384 2020 CA1
verify return:1
depth=0 C = CY, L = Dali, O = MG Freesites Ltd, CN = *.pornhub.com
verify return:1
---
[truncated]
```

I found https://github.com/openssl/openssl/issues/12731#issuecomment-681949206, where matt implies the server did not acknowledge the SNI extension. 

However in my case, I never sent the SNI extension, though the server did give it's cert and everything else proceeded as expected:

![image](https://github.com/openssl/openssl/assets/6680615/c6b7bae1-c599-42c8-a0ba-4e6e787b2770)

Just want to understand the meaning of `Can't use SSL_get_servername`.

Thanks. If you need the pcap or similar I'd be happy to attach it.

**Comment by t8m** [comment, marked as answer]
It means there is no SNI extension used.

**Reply by ckcr4lyf** [reply]
Thanks for the clarification.
